kernel/thread.c

```diff
@@ -381,7 +440,11 @@ void z_setup_new_thread(struct k_thread *new_thread,
 	k_spin_unlock(&lock, key);
 #endif
 #ifdef CONFIG_THREAD_NAME
-	new_thread->name = name;
+	if (name) {
+		strncpy(new_thread->name, name, CONFIG_THREAD_MAX_NAME_LEN);
```

Comment:
Might as well pass CONFIG_THREAD_MAX_NAME_LEN-1 here to avoid double-writing the last byte.